ibis/backends/duckdb/__init__.py

```diff
@@ -191,6 +191,10 @@ def _get_schema_using_query(self, query: str) -> sch.Schema:
         """Return an ibis Schema from a DuckDB SQL string."""
         return sch.Schema.from_tuples(self._metadata(query))
 
+    def _register_in_memory_table(self, table_op):
+        df = table_op.data.to_frame()
+        self.con.execute("register", (table_op.name, df))
```

Comment:
It might be worth having a conversation about how to best expose functionality like this from duckdb through the sqlalchemy layer? Engine signals perhaps?